solver/src/settlement/settlement_encoder.rs

```diff
@@ -342,6 +346,41 @@ impl SettlementEncoder {
     }
 }
 
+fn reduce_unwraps(
+    mut unwraps: Vec<UnwrapWethInteraction>,
+    native_asset_balance: U256,
+) -> Vec<UnwrapWethInteraction> {
+    // TODO - Given the context of this vector,
+    //  I don't see any reason this list will ever contain more than one entry.
+    // We sort by amounts to remove the most possible.
+    unwraps.sort_by_key(|unwrap| unwrap.amount);
+    let index = index_with_cumulative_sum_not_exceeding(
+        &unwraps.iter().map(|u| u.amount).collect::<Vec<U256>>(),
+        native_asset_balance,
+    );
+    if let Some(cut_off) = index {
+        return unwraps[cut_off + 1..].to_vec();
```

Comment:
Yes this is the plan here for this PR as well I think we should get rid of the vector of unwraps since we already know it's only ever going to be one entry.